### ISSUE TITLE
Should hlint warn about unnecessary OPTIONS_GHC

### ISSUE THREAD

**Issue by expipiplus1** [ndmitchell/hlint#306]
For example should it warn about the redundant line here because there are no unticked promoted constructors in this file?

```haskell
{-# OPTIONS_GHC -fno-warn-unticked-promoted-constructors #-}
main = pure ()
```

**Comment by ndmitchell** [comment]
Certainly an interesting idea. However, I think there are too many GHC warnings, and their precise semantics are too vague, so I don't think this is a sensible route to take.

I do have a plan to be able to declare a list of permissible `OPTIONS_GHC` flags, which is slightly in this area, and would "ban" people using `-w` or similar. In some cases that will have a similar effect, but certainly not all.

**Comment by ndmitchell** [comment]
See #288 for the restrictions. Other than that, I don't want HLint to know about GHC flags - they are too ill specified.
